packages/@aws-cdk/aws-apigateway/lib/authorizer.ts

```diff
@@ -1,14 +1,29 @@
-import { Resource } from '@aws-cdk/core';
+import { Construct, Resource } from '@aws-cdk/core';
 import { AuthorizationType } from './method';
 import { RestApi } from './restapi';
 
+const AUTHORIZER_SYMBOL = Symbol.for('@aws-cdk/aws-apigateway.Authorizer');
+
 /**
  * Base class for all custom authorizers
  */
 export abstract class Authorizer extends Resource implements IAuthorizer {
+  /**
+   * Return whether the given object is an Authorizer.
+   */
+  public static isAuthorizer(x: any): x is Authorizer {
+    return x !== null && typeof(x) === 'object' && AUTHORIZER_SYMBOL in x;
+  }
+
   public readonly abstract authorizerId: string;
   public readonly authorizationType?: AuthorizationType = AuthorizationType.CUSTOM;
 
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    Object.defineProperty(this, AUTHORIZER_SYMBOL, { value: true });
```

Comment:
```suggestion
    Object.defineProperty(this, AUTHORIZER_SYMBOL, { configurable: false, enumerable: false, value: true });
```

Reply:
These are both the defaults as well. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty